Makefile

```diff
@@ -0,0 +1,29 @@
+all: build check
+
+check: fmt vet lint ## Check your code
+
+revendor:
+	rm glide.lock
+	glide install --strip-vendor
+	glide-vc --use-lock-file --no-tests --only-code
+
+terraform:
+	hack/get-terraform.sh
+
+build: ## build installer binary
+	hack/build.sh
+
+#test: # Run unit test
+#	go test -race -cover ./cmd/...
+
+lint: ## Go lint your code
+	hack/go-lint.sh -min_confidence 0.3 $(go list -f '{{ .ImportPath }}' ./...)
+
+fmt: ## Go fmt your code
+	hack/go-fmt.sh
+
+vet: ## Apply go vet to all go files
+	hack/go-vet.sh ./...
+
+help:
+	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
This doesn't mass my personal "utility outweighs complexity" threshold, although other maintainers may feel differently.  Can we just do something like [this][1]?

[1]: https://github.com/kubernetes-sigs/cri-o/blob/v1.11.6/Makefile#L44-L55

Reply:
I kinda like it. By writing one thing you automatically get two:
- code comments
- runtime help

So no need to maintain a part of Makefile responsible for `help` target.

Reply:
> By writing one thing you automatically get two:
>
> * code comments
> * runtime help

Runtime help is useful for users who don't read `Makefile`.  I don't think the comment role is important though, especially with a `Makefile` as short and simple as this.

A hard-coded approach to `help` is also harder to get wrong (e.g. [this][1], I think?).  And you'll have to expand your regexp to allow slashes for `bin/terraform`.

I also don't know how portable your ANSI color escapes are (e.g. does Windows stock terminal support them?).  I'd rather drop the terminal escapes.

But if you're really committed, I'll `/lgtm` this approach ;).

[1]: https://github.com/openshift/installer/pull/385#discussion_r221975846